src/Confluent.Kafka/Impl/LibRdKafka.cs

```diff
@@ -160,6 +160,7 @@ static LibRdKafka()
             _consumer_close = NativeMethods.rd_kafka_consumer_close;
             _assign = NativeMethods.rd_kafka_assign;
             _assignment = NativeMethods.rd_kafka_assignment;
+            _offsets_store = NativeMethods.rd_kafka_offsets_store;
```

Comment:
This API was added in librdkafka 0.11.0, if we want to be compatible with older librdkafka versions we might want to conditionalize this, yesno?

Reply:
Yes, I agree on your suggestion. Let me think through it.

Reply:
@edenhill :  I understood the part that this feature is only available from version 0.11.0. 

As I am not familiar with the direction on how LibRdKafka version dependency is managed. Usually any new features such as store offsets should be merged into a future version of Confluent.Kafka and lift the minVersion of LibRdKafka to 0.11.0. I mean this feature will be merged to master whenever it is safe to lift minVersion of LibRdKafka to 0.11.0. There must be something I am missing when you asked for conditionalize code.

When you said conditionalize this (_offsets_store = NativeMethods.rd_kafka_offsets_store;), I didn't fully understand your exact intention. Are you asking me to add a pre validation inside SafeKafkaHandle.cs StoreOffsets method to check if LibRdKafka version is less than 0.11.0 and throw something like NotSupportedException with a meaningful message?

Reply:
All valid questions @vinodres, I'll leave @mhowlett to answer them :)

Reply:
For now, this could be via Librdkafka.cs (`const long minVersion = 0x00110000`), no code is doing checking (we surely should)

We could provide some more fined process for this. Either check at each call what is the version loaded (and throwing exception when version is not ok as @vinodres  stated), either do this via attibute `[MinVersion(0x00110000)]` (cleaner, but runtime cost would not be negligeable)

I think this will need some thinking in relation with #340, which will allow to more easily load any version of librdkafka

Reply:
Yep, I think we should stick with just enforcing a minimum librdkafka version globally - more fine grained control doesn't seem to have enough benefit to be worth the hassle to me. We do have a situation at the moment where the optimal thing for some people to do is use librdkafka 0.9.5 with Confluent.Kafka 0.11.0 due to regressions in librdkafka, and if we'd enforced 0.11.0 in Confluent.Kafka they couldn't do this. However hopefully this should happen rarely, and it's also not such a big deal to downgrade to Confluent.Kafka 0.9.5.

Reply:
But is it possible to downgrade to v0.9.5 with this dependency on offsets_store that was added in 0.11.0?

Reply:
`GetDelegateForFunctionPointer` (which we will use going forward due to #340) throws an exception in the case of a `IntPtr.Null` which is what it gets if a symbol lookup fails and so the way things are planned to be currently, all the symbols must exist in librdkafka on first `Producer` or `Consumer` construction and there is no way to recover from this. 

But you give me an idea - I could wrap `GetDelegateForFunctionPointer` in another function that catches exceptions and instead returns a new delegate of the appropriate type that throws an exception if it is used.